proto/cmp/services/activity/v1alpha/activity_types.proto

```diff
@@ -0,0 +1,235 @@
+  string context = 1;
+
+  // Errors encountered in the generation of the response. Standardized error list
+  // to be created.
+  repeated string errors = 2;
+
+  // Warnings encountered in the generation of the response. This section might include
+  // information about fields not processed due to incompatibilities with the
+  // inventory system
+  repeated string warnings = 3;
+
+  // Supplier code from the Inventory System for this activity response.
+  string supplier_unit_code = 4;
+
+  // An identifier for external sessions, aiding in tracking and continuity across
+  // sessions.
+  string external_session_id = 5;
+
+  // Activity Category Code
+  string category_code = 6;
+
+  // Category name which describes the category_code
+  string category_name = 7;
+
+  // Type Code
+  string type_code = 8;
+
+  // Type name which describes type_code
+  string type_name = 9;
+
+  // Supplier specific code, matching the supplier code in ProductList and ProductInfo
+  // messages
+  cmp.types.v1alpha.SupplierProductCode supplier_code = 10;
+
+  // Supplier Name
+  string supplier_code_name = 11;
+
+  // product_code
+  cmp.types.v1alpha.ProductCode product_code = 12;
+
+  // Duration range (min, max) of the activity in minutes
+  cmp.types.v1alpha.DurationRange duration_range = 13;
+
+  // Time given for confirmation before the offer expires
+  cmp.types.v1alpha.Duration max_confirmation_duration = 14; 
+
+  // Allow Free Sale
+  bool allow_free_sale = 15;
+
+  // The ability to confirm instantly
+  bool instant_confirmation = 16;
+
+  // Can be delivered instantly
+  bool instant_delivery = 17;
+
+  // Check availability
+  bool availability_required = 18;
+  
+  // Availability Type
+  string availability_type = 19;
+
+  // Delivery Formats; such as QR Code, NFT, Ticket, ... etc
+  repeated string delivery_formats = 20;
+
+  // Delivery Methods; such as Email, SMS, Post, ... etc
+  repeated string delivery_methods = 21;
+
+  // Redemption method
+  string redemption_method = 22;
+
+}
+
+
+message ActivityUnitInfo {
+  // Schedule
+  cmp.types.v1alpha.DateTimeRange schedule = 1; 
+
+  // Description of the activity unit
+  string descriptive_text = 2;
+
+  // Unit Code
+  string unit_code = 3;
+
+  // Unit Code Description
+  string unit_code_description = 4;
+
+}
+
+
+message ActivityServiceInfo {
+  // Description of the Service Info
+  string descriptive_text = 1;
+  
+  // Service Code
+  string service_code = 2;
+
+  // Service Code Description
+  string service_code_description = 3;
+
+  // Included items in the activity service
+  repeated string included = 4;
+
+  // Encluded items in the activity service
+  repeated string excluded = 5;
+}
+
+message ActivityZoneInfo {
+  // Sub Zone Info
+  ActivitySubZoneInfo sub_zone_info = 1;
+
+  // Unit Code
+  string unit_code = 2;
+}
+
+message ActivitySubZoneInfo {
+  // Geo tree type, representted by Country, Region, and City_or_Resort.
+  repeated cmp.types.v1alpha.GeoTree ref_points = 1;
```

Comment:
Is the field name correct? It doesn't make an intuitive sense to me. What's "points" with a GeoTree object?

Also, I scanned through the json and didn't find any list of "ref_points". All of them have a single "ref_point" inside them and that "ref_point" is a GeoTree object.

Reply:
this section describes various pick-up/drop-off points in different resorts for one excursion. As it describes a single subzone, it does not need to be repeated.

Reply:
regarding the naming, yes we can change it. I suggest geo_tree like other messages.

regarding the repeated, yes in the json ref_points has a single object ref_point , which is useless, that's why we removed this single object. And if you scroll down in the json you will find that ref_point object sometimes repeated and sometimes not. That's why I put repeated.